### ISSUE TITLE
Installation problems

### ISSUE THREAD

**Issue by blaisn** [doberkofler/web_plsql#4]
Hi,
Starting from en empty project, I followed the instructions:
Create and move to a new directory
Create a new npm project (npm init)
Install package (npm install web_plsql)
I got some errors:
C:\sources\web-plsql>npm install web_plsql --save

> oracledb@3.1.1 install C:\sources\web-plsql\node_modules\web_plsql\node_modules\oracledb
> node package/install.js
oracledb ********************************************************************************
oracledb ** Node-oracledb 3.1.1 installed for Node.js 10.15.3 (win32, x64)
oracledb **
oracledb ** To use node-oracledb:
oracledb ** - Oracle Client libraries (64-bit) must be in your PATH environment variable
oracledb ** - To get libraries, install an Instant Client Basic or Basic Light package from
oracledb **   https://www.oracle.com/technetwork/topics/winx64soft-089540.html
oracledb ** - A Microsoft Visual Studio Redistributable suitable for your Oracle Client library version must be available
oracledb **   See https://oracle.github.io/node-oracledb/INSTALL.html for details
oracledb **
oracledb ** Installation instructions: https://oracle.github.io/node-oracledb/INSTALL.html
oracledb ********************************************************************************

> web_plsql@0.1.2 install C:\sources\web-plsql\node_modules\web_plsql
> npm run build

> web_plsql@0.1.2 build C:\sources\web-plsql\node_modules\web_plsql
> tsc --project ./tsconfig.src.json

src/cgi.ts:5:16 - error TS2307: Cannot find module 'os'.

5 import os from 'os';
                 ~~~~

src/cgi.ts:6:19 - error TS2307: Cannot find module 'url'.

6 import {URL} from 'url';
                    ~~~~~

src/cgi.ts:7:21 - error TS7016: Could not find a declaration file for module 'express'. 'C:/sources/web-plsql/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

7 import express from 'express';
                      ~~~~~~~~~

src/errorPage.ts:5:18 - error TS2307: Cannot find module 'util'.

5 import util from 'util';
                   ~~~~~~

src/errorPage.ts:6:20 - error TS7016: Could not find a declaration file for module 'escape-html'. 'C:/sources/web-plsql/node_modules/escape-html/index.js' implicitly has an 'any' type.
  Try `npm install @types/escape-html` if it exists or add a new declaration (.d.ts) file containing `declare module 'escape-html';`

6 import escape from 'escape-html';
                     ~~~~~~~~~~~~~

src/errorPage.ts:9:21 - error TS7016: Could not find a declaration file for module 'express'. 'C:/sources/web-plsql/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

9 import express from 'express';
                      ~~~~~~~~~

src/errorPage.ts:49:2 - error TS2584: Cannot find name 'console'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.

49  console.error(output.text);
    ~~~~~~~

src/fileUpload.ts:5:18 - error TS2307: Cannot find module 'path'.

5 import path from 'path';
                   ~~~~~~

src/fileUpload.ts:6:16 - error TS2307: Cannot find module 'fs'.

6 import fs from 'fs';
                 ~~~~

src/fileUpload.ts:7:21 - error TS7016: Could not find a declaration file for module 'express'. 'C:/sources/web-plsql/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

7 import express from 'express';
                      ~~~~~~~~~

src/fileUpload.ts:8:22 - error TS7016: Could not find a declaration file for module 'oracledb'. 'C:/sources/web-plsql/node_modules/web_plsql/node_modules/oracledb/index.js' implicitly has an 'any' type.
  Try `npm install @types/oracledb` if it exists or add a new declaration (.d.ts) file containing `declare module 'oracledb';`

8 import oracledb from 'oracledb';
                       ~~~~~~~~~~

src/fileUpload.ts:109:10 - error TS7006: Parameter 'result' implicitly has an 'any' type.

109             .then(result => {
                      ~~~~~~

src/fileUpload.ts:116:39 - error TS7006: Parameter 'e' implicitly has an 'any' type.

116             }).catch(/* istanbul ignore next */e => {
                                                   ~

src/index.ts:5:17 - error TS2307: Cannot find module 'url'.

5 import url from 'url';
                  ~~~~~

src/index.ts:6:21 - error TS7016: Could not find a declaration file for module 'express'. 'C:/sources/web-plsql/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

6 import express from 'express';
                      ~~~~~~~~~

src/index.ts:7:22 - error TS7016: Could not find a declaration file for module 'oracledb'. 'C:/sources/web-plsql/node_modules/web_plsql/node_modules/oracledb/index.js' implicitly has an 'any' type.
  Try `npm install @types/oracledb` if it exists or add a new declaration (.d.ts) file containing `declare module 'oracledb';`

7 import oracledb from 'oracledb';
                       ~~~~~~~~~~

src/index.ts:13:17 - error TS2580: Cannot find name 'require'. Do you need to install type definitions for node? Try `npm i @types/node` and then add `node` to the types field in your tsconfig.

13 const version = require('../package.json').version;
                   ~~~~~~~

src/index.ts:35:1 - error TS2304: Cannot find name 'exports'.

35 exports = module.exports = webplsql;
   ~~~~~~~

src/index.ts:35:11 - error TS2580: Cannot find name 'module'. Do you need to install type definitions for node? Try `npm i @types/node` and then add `node` to the types field in your tsconfig.

35 exports = module.exports = webplsql;
             ~~~~~~

src/page.ts:6:21 - error TS7016: Could not find a declaration file for module 'express'. 'C:/sources/web-plsql/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

6 import express from 'express';
                      ~~~~~~~~~

src/procedure.ts:5:22 - error TS7016: Could not find a declaration file for module 'oracledb'. 'C:/sources/web-plsql/node_modules/web_plsql/node_modules/oracledb/index.js' implicitly has an 'any' type.
  Try `npm install @types/oracledb` if it exists or add a new declaration (.d.ts) file containing `declare module 'oracledb';`

5 import oracledb from 'oracledb';
                       ~~~~~~~~~~

src/procedure.ts:12:21 - error TS7016: Could not find a declaration file for module 'express'. 'C:/sources/web-plsql/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

12 import express from 'express';
                       ~~~~~~~~~

src/procedureError.ts:17:13 - error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.

17      if (Error.captureStackTrace) {
                  ~~~~~~~~~~~~~~~~~

src/procedureError.ts:18:10 - error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.

18              Error.captureStackTrace(this, ProcedureError);
                      ~~~~~~~~~~~~~~~~~

src/request.ts:5:18 - error TS2307: Cannot find module 'util'.

5 import util from 'util';
                   ~~~~~~

src/request.ts:6:22 - error TS7016: Could not find a declaration file for module 'oracledb'. 'C:/sources/web-plsql/node_modules/web_plsql/node_modules/oracledb/index.js' implicitly has an 'any' type.
  Try `npm install @types/oracledb` if it exists or add a new declaration (.d.ts) file containing `declare module 'oracledb';`

6 import oracledb from 'oracledb';
                       ~~~~~~~~~~

src/request.ts:12:21 - error TS7016: Could not find a declaration file for module 'express'. 'C:/sources/web-plsql/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

12 import express from 'express';
                       ~~~~~~~~~

src/request.ts:58:3 - error TS2584: Cannot find name 'console'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.

58      console.error(`Unable to release database connection\n${e.message}`);
        ~~~~~~~

src/requestError.ts:12:13 - error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.

12      if (Error.captureStackTrace) {
                  ~~~~~~~~~~~~~~~~~

src/requestError.ts:13:10 - error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.

13              Error.captureStackTrace(this, RequestError);
                      ~~~~~~~~~~~~~~~~~

src/stream.ts:1:20 - error TS2307: Cannot find module 'stream'.

1 import stream from 'stream';
                     ~~~~~~~~

src/stream.ts:9:74 - error TS2580: Cannot find name 'Buffer'. Do you need to install type definitions for node? Try `npm i @types/node` and then add `node` to the types field in your tsconfig.

9 export async function streamToBuffer(readable: stream.Readable): Promise<Buffer> {
                                                                           ~~~~~~

src/stream.ts:11:24 - error TS2580: Cannot find name 'Buffer'. Do you need to install type definitions for node? Try `npm i @types/node` and then add `node` to the types field in your tsconfig.

11      const buffers: Array<Buffer> = [];
                             ~~~~~~

src/stream.ts:13:23 - error TS7006: Parameter 'buffer' implicitly has an 'any' type.

13      readable.on('data', buffer => {
                            ~~~~~~

src/stream.ts:18:12 - error TS2552: Cannot find name 'Buffer'. Did you mean 'buffers'?

18              resolve(Buffer.concat(buffers));
                        ~~~~~~

  src/stream.ts:11:9
    11          const buffers: Array<Buffer> = [];
                      ~~~~~~~
    'buffers' is declared here.

src/stream.ts:21:24 - error TS7006: Parameter 'err' implicitly has an 'any' type.

21      readable.on('error', err => {
                             ~~~

src/trace.ts:5:16 - error TS2307: Cannot find module 'fs'.

5 import fs from 'fs';
                 ~~~~

src/trace.ts:6:18 - error TS2307: Cannot find module 'path'.

6 import path from 'path';
                   ~~~~~~

src/trace.ts:7:20 - error TS7016: Could not find a declaration file for module 'mkdirp'. 'C:/sources/web-plsql/node_modules/mkdirp/index.js' implicitly has an 'any' type.
  Try `npm install @types/mkdirp` if it exists or add a new declaration (.d.ts) file containing `declare module 'mkdirp';`

7 import mkdirp from 'mkdirp';
                     ~~~~~~~~

src/trace.ts:8:18 - error TS2307: Cannot find module 'util'.

8 import util from 'util';
                   ~~~~~~

src/trace.ts:9:21 - error TS7016: Could not find a declaration file for module 'express'. 'C:/sources/web-plsql/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

9 import express from 'express';
                      ~~~~~~~~~

src/trace.ts:41:5 - error TS2584: Cannot find name 'console'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.

41                      console.log(`Tracing to the directory "${this._directory}" is enabled.`);
                        ~~~~~~~

src/trace.ts:45:4 - error TS2584: Cannot find name 'console'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.

45              console.error(`Unable to create new trace directory "${this._directory}"`, e);
                ~~~~~~~

src/trace.ts:70:149 - error TS2339: Property 'originalUrl' does not exist on type 'TCustomRequest'.

70      appendSync(path.join(this._directory, 'trace.log'), `${customRequest.uniqueRequestID.toString().padEnd(10)} - ${getTimestamp()} - ${customRequest.originalUrl}\n`);
                                                                                                                                                          ~~~~~~~~~~~

src/trace.ts:156:3 - error TS2584: Cannot find name 'console'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.

156     console.error(`Unable to write to trace file "${filename}"`, e, text);
        ~~~~~~~


Found 45 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! web_plsql@0.1.2 build: `tsc --project ./tsconfig.src.json`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the web_plsql@0.1.2 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\blais_n\AppData\Roaming\npm-cache\_logs\2019-03-06T21_09_23_511Z-debug.log
npm WARN request-promise-native@1.0.7 requires a peer of request@^2.34 but none is installed. You must install peer dependencies yourself.
npm WARN request-promise-core@1.1.2 requires a peer of request@^2.34 but none is installed. You must install peer dependencies yourself.
npm WARN web-plsql@1.0.0 No description
npm WARN web-plsql@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! web_plsql@0.1.2 install: `npm run build`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the web_plsql@0.1.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\blais_n\AppData\Roaming\npm-cache\_logs\2019-03-06T21_09_23_934Z-debug.log

To manually eliminate these errors, I had to manually install:
npm install -g typescript
npm install express
npm install escape-html
npm install @types/express
npm install @types/oracledb
npm install @types/escape-html
npm install @types/mkdirp

Use of node 10.15.3

Did I miss something?

**Comment by ganesh-latotools** [comment]
Thanks for the tip at the end of your comment. 
These have helped me.
`
npm install -g typescript
npm install express
npm install escape-html
npm install @types/express
npm install @types/oracledb
npm install @types/escape-html
npm install @types/mkdirp
`

**Comment by doberkofler** [comment]
Thank you for the information. This dependency problem should now be resolved in 0.2.0.
